native/cocos/2d/renderer/RenderDrawInfo.cpp

```diff
@@ -52,55 +52,50 @@ void RenderDrawInfo::changeMeshBuffer() {
 
 gfx::InputAssembler* RenderDrawInfo::requestIA(gfx::Device* device) {
     CC_ASSERT(_drawInfoAttrs._isMeshBuffer && _drawInfoAttrs._drawInfoType == RenderDrawInfoType::COMP);
-    if (!_iaPool) {
-        _iaPool = ccnew ccstd::vector<gfx::InputAssembler*>;
-    }
-    if (_nextFreeIAHandle >= _iaPool->size()) {
-        initIAInfo(device);
-    }
-    auto* ia = (*_iaPool)[_nextFreeIAHandle++];
-    ia->setFirstIndex(getIndexOffset());
-    ia->setIndexCount(getIbCount());
-    return ia;
+    initIAInfo(device);
+    return _ia;
 }
 
 void RenderDrawInfo::uploadBuffers() {
     CC_ASSERT(_drawInfoAttrs._isMeshBuffer && _drawInfoAttrs._drawInfoType == RenderDrawInfoType::COMP);
     if (_drawInfoAttrs._vbCount == 0 || _drawInfoAttrs._ibCount == 0) return;
+    bool needUpdateIA = false;
+
     uint32_t size = _drawInfoAttrs._vbCount * 9 * sizeof(float); // magic Number
-    gfx::Buffer* vBuffer = _iaInfo->vertexBuffers[0];
-    vBuffer->resize(size);
-    vBuffer->update(_vDataBuffer);
-    gfx::Buffer* iBuffer = _iaInfo->indexBuffer;
-    uint32_t iSize = _drawInfoAttrs._ibCount * 2;
-    iBuffer->resize(iSize);
-    iBuffer->update(_iDataBuffer);
+    const auto &vBuffers = _ia->getVertexBuffers();
+    if (!vBuffers.empty()) {
+        gfx::Buffer* vBuffer = vBuffers[0];
+        needUpdateIA |= vBuffer->resize(size);
+        vBuffer->update(_vDataBuffer);
+    }
+
+    gfx::Buffer* iBuffer = _ia->getIndexBuffer();
+    if (iBuffer != nullptr) {
+        uint32_t iSize = _drawInfoAttrs._ibCount * 2;
+        needUpdateIA |= iBuffer->resize(iSize);
+        iBuffer->update(_iDataBuffer);
+    }
+
+    if (needUpdateIA) {
+        _ia->update();
+    }
 }
 
-void RenderDrawInfo::resetMeshIA() {
+void RenderDrawInfo::resetMeshIA() const {
     CC_ASSERT(_drawInfoAttrs._isMeshBuffer && _drawInfoAttrs._drawInfoType == RenderDrawInfoType::COMP);
-    _nextFreeIAHandle = 0;
 }
```

Comment:
It seems this function can be removed?

Reply:
Yes, removing the interface will cause a compile error when running test cases. So I added back and ignore it in swig config. It could be removed after jsb automatically updated.

Reply:
> removing the interface will cause a compile error when running test cases.

It is because it uses a precompiled editor to run test case. It is different from other CI. Other CI will use latest engine codes.